python/cugraph/pregel/traversal/bfs_bsp.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import cudf
+from collections import OrderedDict
+
+
+def bfs_df_pregel(df, start, src_col='src', dst_col='dst', copy_data=True):
```

Comment:
Naming : 
- `pregel` is tightly connected to google's system. Should we use `bsp` or something more generic?
- is `_df` necessary?

API: the `Graph` object takes care of renumbering, casting and input checking. Since we are not using it in this analytics, we may want to add similar features here in the long run. 
Do we expect all pregerl/bsp functions to operate on a DF edge list directly instead of a `(Di)Graph`?

Reply:
I would rather use BSP.  Pregel did start as a Google thing but the name is not used within Spark.

I would like to add an additional function (wrapper) that takes the new Graph object.

Reply:
It should be _df to avoid collisions